filters/tracing/tag.go

```diff
@@ -38,22 +46,36 @@ func (s tagSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
 		return nil, filters.ErrInvalidFilterParameters
 	}
 
-	return tagFilter{
+	return &tagFilter{
+		tagFromResponse: s.typ == filters.TracingTagFromResponseName,
+
 		tagName:  tagName,
 		tagValue: eskip.NewTemplate(tagValue),
 	}, nil
 }
 
-func (f tagFilter) Request(ctx filters.FilterContext) {
-	req := ctx.Request()
-	span := opentracing.SpanFromContext(req.Context())
-	if span == nil {
-		return
-	}
+func (f *tagFilter) Request(ctx filters.FilterContext) {
+	if !f.tagFromResponse {
+		span := opentracing.SpanFromContext(ctx.Request().Context())
+		if span == nil {
+			return
+		}
 
-	if v, ok := f.tagValue.ApplyContext(ctx); ok {
-		span.SetTag(f.tagName, v)
+		if v, ok := f.tagValue.ApplyContext(ctx); ok {
+			span.SetTag(f.tagName, v)
+		}
 	}
 }
 
-func (f tagFilter) Response(filters.FilterContext) {}
+func (f *tagFilter) Response(ctx filters.FilterContext) {
+	if f.tagFromResponse {
+		span := opentracing.SpanFromContext(ctx.Request().Context())
+		if span == nil {
+			return
+		}
+
+		if v, ok := f.tagValue.ApplyContext(ctx); ok {
+			span.SetTag(f.tagName, v)
+		}
+	}
```

Comment:
Could you move this to a helper `func (f *tagFilter) setTag(ctx filters.FilterContext)` please?